src/GRA.Controllers/AvatarController.cs

```diff
@@ -59,12 +59,70 @@ public async Task<IActionResult> Index()
                 .GroupBy(_ => _.GroupId)
                 .Select(_ => _.ToList())
                 .ToList();
+            var usersresult = _avatarService.GetUserUnlockBundles().Result;
+            var userbundles = usersresult.Item1;
+            var hasbeenviewed = usersresult.Item2;
+            var bundles = new AvatarBundleJsonModel
+            {
+                Bundles = _mapper
+                .Map<List<AvatarBundleJsonModel.AvatarBundle>>(userbundles)
+            };
             var viewModel = new AvatarViewModel
             {
                 LayerGroupings = layerGroupings,
+                Bundles = userbundles,
+                ViewedBundles = hasbeenviewed,
                 DefaultLayer = userWardrobe.First(_ => _.DefaultLayer).Id,
                 ImagePath = _pathResolver.ResolveContentPath($"site{GetCurrentSiteId()}/avatars/"),
-                AvatarPiecesJson = Newtonsoft.Json.JsonConvert.SerializeObject(model)
+                AvatarPiecesJson = Newtonsoft.Json.JsonConvert.SerializeObject(model),
+                AvatarBundlesJson = Newtonsoft.Json.JsonConvert.SerializeObject(bundles)
+            };
+
+            var userAvatar = await _avatarService.GetUserAvatarAsync();
+            if (userAvatar.Count == 0)
+            {
+                viewModel.NewAvatar = true;
+            }
+            return View(viewModel);
+        }
+
+        public async Task<IActionResult> BundleIndex()
+        {
+            var userWardrobe = await _avatarService.GetUserWardrobeAsync();
+            if (userWardrobe.Count == 0)
+            {
+                ShowAlertDanger("Avatars have not been configured.");
+                _logger.LogError("User {id} tried to customize their avatar but avatars have not been configured!",
+                    GetActiveUserId());
+                return RedirectToAction(nameof(HomeController.Index), HomeController.Name);
+            }
+
+            var model = new AvatarJsonModel
+            {
+                Layers = _mapper
+                .Map<ICollection<AvatarJsonModel.AvatarLayer>>(userWardrobe)
+            };
+            var layerGroupings = userWardrobe
+                .GroupBy(_ => _.GroupId)
+                .Select(_ => _.ToList())
+                .ToList();
+            var usersresult = _avatarService.GetUserUnlockBundles().Result;
+            var userbundles = usersresult.Item1;
+            var hasbeenviewed = usersresult.Item2;
+            var bundles = new AvatarBundleJsonModel
+            {
+                Bundles = _mapper
+                .Map<List<AvatarBundleJsonModel.AvatarBundle>>(userbundles)
+            };
+            var viewModel = new AvatarViewModel
+            {
+                LayerGroupings = layerGroupings,
+                Bundles = userbundles,
+                ViewedBundles = hasbeenviewed,
+                DefaultLayer = 13,
```

Comment:
Why is this 13? This doesn't seem like it should be hardcoded.

Reply:
All the layers are objects with their specific attributes but, when we get to the bundles tab, we dont have a layer object created for it because its not its own layer. Its a different layer that represents all the other layers, at the same time

Reply:
It's not always going to necessarily be 13, though. We need to figure out a way to pass the 13 to this code from where it is assigned or have a way to look the value up.

Reply:
What if we did layerGroupings.count +1 ?

Reply:
Or 99?

Reply:
I think 999 might be the safest bet?